docs/static/persistent-queues.asciidoc

```diff
@@ -1,44 +1,245 @@
+
+[[pqcheck]]
+===== `pqcheck` utility
+
+Run the `pqcheck` utility to identify which persistent queue--or queues--have been corrupted. 
+
+From LOGSTASH_HOME, run:
+
+[source,txt]
+-----
+bin/pqcheck <queue_folder>
+-----
 
-The following are problems not solved by the persistent queue feature:
+where `<queue_folder>` is the fully qualified path to the persistent queue location.
 
-* Input plugins that do not use a request-response protocol cannot be protected from data loss. For example: tcp, udp, zeromq push+pull, and many other inputs do not have a mechanism to acknowledge receipt to the sender. Plugins such as beats and http, which *do* have an acknowledgement capability, are well protected by this queue.
-* It does not handle permanent machine failures such as disk corruption, disk failure, and machine loss. The data persisted to disk is not replicated.
+The `pqcheck utility` reads through the checkpoint files in the given directory and outputs information about the current state of those files. 
+The utility outputs this information for each checkpoint file:
+
+* Filename
+* Whether or not the page file has been fully acknowledged.
+A fully acknowledged page file indicates that all events have been read and processed.
+* Page number
+* Size of the page file. A page file with a size of 0 results in the output `NOT FOUND`. 
+In this case, run `pqrepair` against the specified queue directory. 
+* Page number
+* First unacknowledged page number (only relevant in the head checkpoint)
+* First unacknowledged event sequence number in the page
+* First event sequence number in the page
+* Number of events in the page
+* Whether or not the page has been fully acknowledged
+
+
+
+*Sample with healthy page file*
+[source,txt]
+-----
+ubuntu@bigger:/usr/share/logstash$ bin/pqcheck /var/lib/logstash/queue/main/
+Using bundled JDK: /usr/share/logstash/jdk
+OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
+Checking queue dir: /var/lib/logstash/queue/main
+checkpoint.1, fully-acked: NO, page.1 size: 67108864
+pageNum=1, firstUnackedPageNum=0, firstUnackedSeqNum=239675, minSeqNum=239675,  elementCount=218241, isFullyAcked=no <1> <2>
+checkpoint.head, fully-acked: NO, page.2 size: 67108864
+pageNum=2, firstUnackedPageNum=0, firstUnackedSeqNum=457916, minSeqNum=457916, elementCount=11805, isFullyAcked=no
+checkpoint.0, fully-acked: NO,<2> page.0 size: 67108864 
+pageNum=0, firstUnackedPageNum=0, firstUnackedSeqNum=176126, minSeqNum=1, elementCount=239674, isFullyAcked=no
+-----
+<1> Numbers indicate message sequence, number of items inside each page file, number of elements in the page, and how far has Logstash progressed through the page file.
+<2> Checkpoints may show `fully-acked: NO` or `fully-acked: YES`.
+After all elements in the page file have been read (events ingested), the status of `fully-acked` will go from `NO` to `YES`.
```

Comment:
Maybe we should fold these "better" comments into the main section explaining the entries in this file?